### This notebook is for importing data from Local folder to PostgreSQL Database

In [8]:
# For Loading the data 

import pandas as pd
from sqlalchemy import create_engine

# Connection string and engine setup
conn_string = 'postgresql://postgres:Binoy%4009@localhost/paintings'
db = create_engine(conn_string)

try:
    # Establish the connection
    with db.connect() as conn:
        files = ['artist', 'canvas_size', 'image_link', 'museum_hours', 'museum', 'product_size', 'subject', 'work']
        
        for file in files:
            try:
                # Load CSV file into DataFrame
                df = pd.read_csv(f'C:/Users/Datacore/Desktop/Paintings Data Analysis/Dataset/{file}.csv')
                # Write DataFrame to SQL table
                df.to_sql(file, con=conn, if_exists='replace', index=False)
                print(f"Successfully loaded {file} into the database.")
            except Exception as e:
                print(f"Failed to load {file}: {e}")

except Exception as e:
    print(f"Error connecting to the database: {e}")


Successfully loaded artist into the database.
Successfully loaded canvas_size into the database.
Successfully loaded image_link into the database.
Successfully loaded museum_hours into the database.
Successfully loaded museum into the database.
Successfully loaded product_size into the database.
Successfully loaded subject into the database.
Successfully loaded work into the database.


In [10]:
# For Validation (No of records)

files = ['artist', 'canvas_size', 'image_link', 'museum_hours', 'museum', 'product_size', 'subject', 'work']

for file in files:
        # Load CSV file into DataFrame
        df = pd.read_csv(f'C:/Users/Datacore/Desktop/Paintings Data Analysis/Dataset/{file}.csv')
        # Print the shape of the DataFrame
        print(f"Shape of {file}: {df.shape}")

Shape of artist: (421, 9)
Shape of canvas_size: (200, 4)
Shape of image_link: (14775, 4)
Shape of museum_hours: (351, 4)
Shape of museum: (57, 9)
Shape of product_size: (110347, 4)
Shape of subject: (6771, 2)
Shape of work: (14776, 5)


In [ ]:
# Here is the SQL Code (Run in Pgadmin4)

DO $$
DECLARE
    tbl_name TEXT;
    row_count INT;
BEGIN
    FOR tbl_name IN
        SELECT table_name
        FROM information_schema.tables
        WHERE table_schema = 'public'
    LOOP
        EXECUTE format('SELECT COUNT(*) FROM %I', tbl_name) INTO row_count;
        RAISE NOTICE '%: % rows', tbl_name, row_count;
    END LOOP;
END $$;

# Output: 
NOTICE:  artist: 421 rows                   
NOTICE:  canvas_size: 200 rows                  
NOTICE:  image_link: 14775 rows                  
NOTICE:  museum_hours: 351 rows                  
NOTICE:  museum: 57 rows                  
NOTICE:  product_size: 110347 rows                  
NOTICE:  subject: 6771 rows                  
NOTICE:  work: 14776 rows                  